ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -95,8 +101,12 @@ S3Connection::ListObjects(const std::string &prefix, std::vector<std::string> &o
     while (continuationToken != "") {
         request.SetContinuationToken(continuationToken);
         outcomes = _s3CrtClient.ListObjectsV2(request);
-        if (!outcomes.IsSuccess())
-            return (1);
+        if (!outcomes.IsSuccess()) {
+            if (toErrno.find(resCode) != toErrno.end())
```

Comment:
You need to update the resCode in this loop. You are using the code from the first request.